controllers/storagecluster/reconcile.go

```diff
@@ -496,6 +497,38 @@ func (r *StorageClusterReconciler) reconcilePhases(
 
 			// mark operator upgradeable if and only if all storageclusters are ready
 			if r.clusters.AreOtherStorageClustersReady(instance) {
+				if instance.Spec.AllowRemoteStorageConsumers {
+					scList := &ocsv1alpha1.StorageConsumerList{}
+					err := r.Client.List(r.ctx, scList, &client.ListOptions{Namespace: instance.Namespace})
+					if err != nil {
+						r.Log.Error(err, "Failed to list StorageConsumers")
+						return reconcile.Result{}, err
+					}
+					providerVersion, err := semver.Make(version.Version)
+					if err != nil {
+						r.Log.Error(err, "Failed to parse OCS Version")
+						return reconcile.Result{}, err
+					}
+					for idx := range scList.Items {
+						clientVersion, err := semver.Make(scList.Items[idx].Status.Client.OperatorVersion)
+						if err != nil {
+							r.Log.Error(err, "Failed to parse Client operator version")
+							return reconcile.Result{}, err
+						}
+						// onboarded client operator can be either on same provider operator version or one less in minor version
+						// we need to stop provider operator upgrades for these lagging client operators only
+						// because if provider operator upgrades, that lagging client operator will become unsupported
+						if providerVersion.Minor-1 == clientVersion.Minor {
+							message := fmt.Sprintf("StorageConsumer %q will become unsupported if upgraded", scList.Items[idx].GetName())
```

Comment:
I don't think we want a condition for each individual consumer. Also `StorageConsumer` does not have a version it is the client operator that has a version

I would suggest having a flag to capture the existence of a client that is in the wrong version, exit the loop, and add a general condition. Maybe with the message: "One or more of the connected ODF clients are not up to date"

We can even count the number of clients that prevent upgrade in the loop and then have the message: "X connected ODF clients are not up to date""

Reply:
done.